lib/bolt/plugin/pkcs7.rb

```diff
@@ -0,0 +1,76 @@
+# frozen_string_literal: true
+
+require 'bolt/secret/base'
+require 'fileutils'
+
+module Bolt
+  class Plugin
+    class Pkcs7 < Bolt::Secret::Base
+      def name
+        'pkcs7'
+      end
+
+      def initialize(boltdir, options)
+        require 'openssl'
+        @boltdir = boltdir
+        @options = options || {}
+      end
+
+      def private_key_path
+        path = @options['private-key'] || 'keys/private_key.pkcs7.pem'
+        File.absolute_path(path, @boltdir)
+      end
+
+      def private_key
+        @private_key ||= OpenSSL::PKey::RSA.new(File.read(private_key_path))
+      end
+
+      def public_key_path
+        path = @options['public-key'] || 'keys/public_key.pkcs7.pem'
+        File.absolute_path(path, @boltdir)
+      end
+
+      def public_key
+        @public_key ||= OpenSSL::X509::Certificate.new(File.read(public_key_path))
+      end
+
+      def keysize
+        @options['keysize'] || 2048
+      end
+
+      # The following implementations are intended to be compatible with hiera-eyaml
+      def encrypt_value(plaintext)
+        cipher = OpenSSL::Cipher::AES.new(256, :CBC)
+        OpenSSL::PKCS7.encrypt([public_key], plaintext, cipher, OpenSSL::PKCS7::BINARY).to_der
+      end
+
+      def decrypt_value(ciphertext)
+        pkcs7 = OpenSSL::PKCS7.new(ciphertext)
+        pkcs7.decrypt(private_key, public_key)
+      end
+
+      def secret_createkeys
+        key = OpenSSL::PKey::RSA.new(keysize)
```

Comment:
Logging might be nice here. Also would users expect keys to be overwritten if they exist?

Reply:
Should we only log if we're replacing file? Should it refuse to replace files without confirmation or a flag?